src/posix.c

```diff
@@ -239,9 +239,13 @@ int p_mmap(git_map *out, size_t len, int prot, int flags, int fd, git_off_t offs
 	out->data = NULL;
 	out->len = 0;
 
-	if ((prot & GIT_PROT_WRITE) && ((flags & GIT_MAP_TYPE) == GIT_MAP_SHARED)) {
-		giterr_set(GITERR_OS, "Trying to map shared-writeable");
-		return -1;
+	if (prot & GIT_PROT_WRITE) {
+		if ((p_lseek(fd, offset, SEEK_SET) < 0) || p_write(fd, out->data, len)) {
+			giterr_set(GITERR_OS, "index write failed");
```

Comment:
Why does this mention an index? The problem is trying to user mmap emulation for writable pages, which is what we don't support. Whether it's the indexer or anything else is irrelevant and having this error message doesn't explain _why_ it failed.